lib/main/atom/codefix/codefixProvider.ts

```diff
@@ -110,4 +74,20 @@ export class CodefixProvider implements IntentionsProvider {
     this.supportedFixes.set(client, codes)
     return codes
   }
+
+  async applyFix(fix: protocol.CodeAction) {
+    for (const f of fix.changes) {
+      const {buffer, isOpen} = await this.getTypescriptBuffer(f.fileName)
+
+      buffer.buffer.transact(() => {
+        for (const edit of f.textChanges) {
+          buffer.buffer.setTextInRange(spanToRange(edit), edit.newText)
+        }
+      })
+
+      if (!isOpen) {
+        ;(buffer.buffer.save() as any).then(() => buffer.buffer.destroy())
```

Comment:
You could `await` here for the `save()`. You won't need to fight with the typings that way.

Reply:
I've thought about that, but decided against, since we don't need to wait for one buffer to be destroyed to work on another.